### ISSUE TITLE
Classical register instructions for simulators

### ISSUE THREAD

**Issue by mont-40** [Qiskit/qiskit#6939]
### What is the expected behavior?

`QuantumCircuit`s are very rich in operations on qubits, but lack operations on classical bits. The feature request are classical operations (logic gates) on classical bits, such as NOT/CNOT (acting "in-place" on the classical register) that are part of the `QuantumCircuit` itself.

### Reasoning

- Simple post-processing of measurement outcomes may be done as part of `QuantumCircuit` simulations
- Combined measurements may be used for classical controlled gates in the simulator (e.g. only do some `Gate` if `cbit0` and `cbit1` are both 1).

**Comment by HuangJunye** [comment]
Are we still planning to fix this issue for 0.20?

**Comment by jakelishman** [comment]
No, this is dynamics circuit work, so it's pushed til at least 0.21.

**Comment by jakelishman** [comment]
Closing as stale - the existence of the `qiskit.circuit.classical` module probably covers the original ask here.
